buildconfig/CMake/WindowsSetup.cmake

```
@@ -59,10 +59,16 @@ if(CONDA_ENV)
   # switch off all optimzations
   set(_conda_debug_cfg_name DebugWithRelRuntime)
   string(TOUPPER ${_conda_debug_cfg_name} _conda_debug_cfg_name_upper)
+  # Set language flags for Visual studio and ninja generators
+  if(CCACHE_FOUND)
+    set(_language_flags "/Z7 /Ob0 /Od /RTC1")
+  else()
+    set(_language_flags "/Zi /Ob0 /Od /RTC1")
+  endif()
```

```suggestion
  # for a single-config generator like Ninja, aim to use CCache and Debug on Clion. The Zi flag is not supported by 
  CCache so we need to use Z7 to allow debugging.
  if(CCACHE_FOUND)
    set(_language_flags "/Z7 /Ob0 /Od /RTC1")
  else()  # For multi-config generators like MSVC, Ccache is not used, so Zi flag can be used for debugging
    set(_language_flags "/Zi /Ob0 /Od /RTC1")
  endif()
```

Feel free to adjust!